### ISSUE TITLE
Could not locate libsymengine

### ISSUE THREAD

**Issue by qua4tre** [symengine/SymEngine.jl#130]
On a fresh install of SymEngine, no other packages installed:
```julia
julia> Pkg.add("SymEngine")
INFO: Installing BinaryProvider v0.3.3
INFO: Installing Compat v1.0.0
INFO: Installing RecipesBase v0.3.1
INFO: Installing SHA v0.5.7
INFO: Installing SymEngine v0.4.1
INFO: Building SymEngine
Info: Downloading https://github.com/JuliaMath/MPFRBuilder/releases/download/v4.0.1/build.jl to /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/downloads/build_36ed4f47426eea41a2576f1c57b4dbbef552c42eea819f8e6c590d394fef049b.jl...
[13:31:45] #####################################################################[13:31:45] #####################################################################[13:31:45] ######################################################################## 100.0%
Info: Downloading https://github.com/JuliaMath/MPFRBuilder/releases/download/v4.0.1/MPFRBuilder.x86_64-linux-gnu.tar.gz to /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/downloads/MPFRBuilder.x86_64-linux-gnu.tar.gz...
[13:31:49] #####################################################################[13:31:49] #####################################################################[13:31:49] #                                                                    [13:31:49] #                                                                    [13:31:49] ##                                                                   [13:31:49] ##                                                                   [13:31:49] ####                                                                 [13:31:49] ####                                                                 [13:31:49] #####                                                                [13:31:49] #####                                                                [13:31:49] ######                                                               [13:31:49] #######                                                              [13:31:49] ########                                                             [13:31:49] ########                                                             [13:31:49] #########                                                            [13:31:49] #########                                                            [13:31:49] ###########                                                          [13:31:49] ###########                                                          [13:31:49] ############                                                         [13:31:49] ############                                                         [13:31:49] ##############                                                       [13:31:49] ##############                                                       [13:31:49] ###############                                                      [13:31:49] ###############                                                      [13:31:49] ################                                                     [13:31:49] ################                                                     [13:31:49] ##################                                                   [13:31:50] ##################                                                   [13:31:50] ###################                                                  [13:31:50] ###################                                                  [13:31:50] #####################                                                [13:31:50] #####################                                                [13:31:50] ######################                                               [13:31:50] ######################                                               [13:31:50] #######################                                              [13:31:50] ########################                                             [13:31:50] #########################                                            [13:31:50] #########################                                            [13:31:50] ##########################                                           [13:31:50] ##########################                                           [13:31:50] ############################                                         [13:31:50] ############################                                         [13:31:50] #############################                                        [13:31:50] #############################                                        [13:31:50] ##############################                                       [13:31:50] ###############################                                      [13:31:50] ################################                                     [13:31:50] ################################                                     [13:31:50] #################################                                    [13:31:50] #################################                                    [13:31:50] ###################################                                  [13:31:50] ###################################                                  [13:31:50] ####################################                                 [13:31:50] ####################################                                 [13:31:50] ######################################                               [13:31:50] ######################################                               [13:31:50] #######################################                              [13:31:50] #######################################                              [13:31:50] ########################################                             [13:31:50] ########################################                             [13:31:50] ##########################################                           [13:31:50] ##########################################                           [13:31:50] ###########################################                          [13:31:50] ###########################################                          [13:31:50] #############################################                        [13:31:50] #############################################                        [13:31:50] ##############################################                       [13:31:50] ##############################################                       [13:31:50] ###############################################                      [13:31:50] ################################################                     [13:31:50] #################################################                    [13:31:50] #################################################                    [13:31:50] ##################################################                   [13:31:50] ##################################################                   [13:31:50] ####################################################                 [13:31:50] ####################################################                 [13:31:50] #####################################################                [13:31:50] #####################################################                [13:31:50] #######################################################              [13:31:50] #######################################################              [13:31:50] ########################################################             [13:31:50] ########################################################             [13:31:50] #########################################################            [13:31:50] #########################################################            [13:31:50] ###########################################################          [13:31:50] ###########################################################          [13:31:50] ############################################################         [13:31:50] ############################################################         [13:31:50] ##############################################################       [13:31:50] ##############################################################       [13:31:50] ###############################################################      [13:31:50] ###############################################################      [13:31:50] ################################################################     [13:31:50] #################################################################    [13:31:50] ##################################################################   [13:31:50] ##################################################################   [13:31:50] ###################################################################  [13:31:50] ###################################################################  [13:31:50] #####################################################################[13:31:50] #####################################################################[13:31:50] #####################################################################[13:31:50] #####################################################################[13:31:50] #####################################################################[13:31:50] ######################################################################## 100.0%
Info: Downloading https://github.com/isuruf/GMPBuilder/releases/download/v6.1.2-1/build_GMP.v6.1.2.jl to /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/downloads/build_9bd8d9078c2a9c9a6451f6850ec45c925727afc395b5e4740153707826cd7439.jl...
[13:31:51] #####################################################################[13:31:51] #####################################################################[13:31:51] ######################################################################## 100.0%
Info: Downloading https://github.com/isuruf/GMPBuilder/releases/download/v6.1.2-1/GMP.v6.1.2.x86_64-linux-gnu.tar.gz to /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/downloads/GMP.v6.1.2.x86_64-linux-gnu.tar.gz...
[13:31:52] #####################################################################[13:31:52] #####################################################################[13:31:52] ##                                                                   [13:31:52] ##                                                                   [13:31:52] ####                                                                 [13:31:53] ####                                                                 [13:31:53] #######                                                              [13:31:53] #######                                                              [13:31:53] #########                                                            [13:31:53] #########                                                            [13:31:53] ###########                                                          [13:31:53] ############                                                         [13:31:53] ##############                                                       [13:31:53] ##############                                                       [13:31:53] ################                                                     [13:31:53] ################                                                     [13:31:53] ###################                                                  [13:31:53] ###################                                                  [13:31:53] #####################                                                [13:31:53] #####################                                                [13:31:53] ########################                                             [13:31:53] ########################                                             [13:31:53] ##########################                                           [13:31:53] ##########################                                           [13:31:53] #############################                                        [13:31:53] #############################                                        [13:31:53] ###############################                                      [13:31:53] ###############################                                      [13:31:53] #################################                                    [13:31:53] ##################################                                   [13:31:53] ####################################                                 [13:31:53] ####################################                                 [13:31:53] ######################################                               [13:31:53] ######################################                               [13:31:53] #########################################                            [13:31:53] #########################################                            [13:31:53] ###########################################                          [13:31:53] ###########################################                          [13:31:53] ##############################################                       [13:31:53] ##############################################                       [13:31:53] ################################################                     [13:31:53] ################################################                     [13:31:53] ##################################################                   [13:31:53] ###################################################                  [13:31:53] #####################################################                [13:31:53] #####################################################                [13:31:53] #######################################################              [13:31:53] #######################################################              [13:31:53] ##########################################################           [13:31:53] ##########################################################           [13:31:53] ############################################################         [13:31:53] ############################################################         [13:31:53] ###############################################################      [13:31:53] ###############################################################      [13:31:53] #################################################################    [13:31:53] #################################################################    [13:31:53] #################################################################### [13:31:53] #################################################################### [13:31:53] #####################################################################[13:31:53] #####################################################################[13:31:53] ######################################################################## 100.0%
Info: Downloading https://github.com/symengine/SymEngineBuilder/releases/download/v0.3.0/build_SymEngine.v0.3.0.jl to /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/downloads/build_cd98583c98c386bc8273ae24efb0ab537f365a115d03342b17623d1ebfde5e6a.jl...
[13:31:54] #####################################################################[13:31:54] #####################################################################[13:31:54] ######################################################################## 100.0%
Info: Downloading https://github.com/symengine/SymEngineBuilder/releases/download/v0.3.0/SymEngine.v0.3.0.x86_64-linux-gnu.tar.gz to /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/downloads/SymEngine.v0.3.0.x86_64-linux-gnu.tar.gz...
[13:31:55] #####################################################################[13:31:55] #####################################################################[13:31:55] #                                                                    [13:31:55] #                                                                    [13:31:55] ##                                                                   [13:31:55] ##                                                                   [13:31:55] ###                                                                  [13:31:55] ####                                                                 [13:31:55] #####                                                                [13:31:55] #####                                                                [13:31:55] ######                                                               [13:31:55] ######                                                               [13:31:55] #######                                                              [13:31:55] ########                                                             [13:31:55] #########                                                            [13:31:55] #########                                                            [13:31:55] ##########                                                           [13:31:55] ##########                                                           [13:31:55] ############                                                         [13:31:55] ############                                                         [13:31:55] #############                                                        [13:31:55] #############                                                        [13:31:55] ##############                                                       [13:31:55] ##############                                                       [13:31:55] ################                                                     [13:31:55] ################                                                     [13:31:55] #################                                                    [13:31:55] #################                                                    [13:31:55] ##################                                                   [13:31:55] ##################                                                   [13:31:55] ####################                                                 [13:31:55] ####################                                                 [13:31:55] #####################                                                [13:31:55] #####################                                                [13:31:55] ######################                                               [13:31:55] ######################                                               [13:31:55] ########################                                             [13:31:55] ########################                                             [13:31:55] #########################                                            [13:31:55] #########################                                            [13:31:55] ##########################                                           [13:31:55] ###########################                                          [13:31:55] ############################                                         [13:31:55] ############################                                         [13:31:55] #############################                                        [13:31:55] #############################                                        [13:31:55] ##############################                                       [13:31:55] ###############################                                      [13:31:55] ################################                                     [13:31:55] ################################                                     [13:31:55] #################################                                    [13:31:55] #################################                                    [13:31:55] ###################################                                  [13:31:55] ###################################                                  [13:31:55] ####################################                                 [13:31:55] ####################################                                 [13:31:55] #####################################                                [13:31:56] #####################################                                [13:31:56] #######################################                              [13:31:56] #######################################                              [13:31:56] ########################################                             [13:31:56] ########################################                             [13:31:56] #########################################                            [13:31:56] #########################################                            [13:31:56] ###########################################                          [13:31:56] ###########################################                          [13:31:56] ############################################                         [13:31:56] ############################################                         [13:31:56] #############################################                        [13:31:56] #############################################                        [13:31:56] ###############################################                      [13:31:56] ###############################################                      [13:31:56] ################################################                     [13:31:56] ################################################                     [13:31:56] #################################################                    [13:31:56] #################################################                    [13:31:56] ###################################################                  [13:31:56] ###################################################                  [13:31:56] ####################################################                 [13:31:56] ####################################################                 [13:31:56] #####################################################                [13:31:56] ######################################################               [13:31:56] #######################################################              [13:31:56] #######################################################              [13:31:56] ########################################################             [13:31:56] ########################################################             [13:31:56] ##########################################################           [13:31:56] ##########################################################           [13:31:56] ###########################################################          [13:31:56] ###########################################################          [13:31:56] ############################################################         [13:31:56] ############################################################         [13:31:56] ##############################################################       [13:31:56] ##############################################################       [13:31:56] ###############################################################      [13:31:56] ###############################################################      [13:31:56] ################################################################     [13:31:56] ################################################################     [13:31:56] ##################################################################   [13:31:56] ##################################################################   [13:31:56] ###################################################################  [13:31:56] ###################################################################  [13:31:56] #################################################################### [13:31:56] #################################################################### [13:31:56] #####################################################################[13:31:56] #####################################################################[13:31:56] #####################################################################[13:31:56] #####################################################################[13:31:56] ######################################################################## 100.0%
Info: Downloading https://github.com/isuruf/MPCBuilder/releases/download/v1.1.0/build_mpc.v1.1.0.jl to /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/downloads/build_d800a46181c40fe046dfd51bf0357393bb58e2475b6082065ad0756971ad82eb.jl...
[13:31:57] #####################################################################[13:31:57] #####################################################################[13:31:57] ######################################################################## 100.0%
Info: Downloading https://github.com/isuruf/MPCBuilder/releases/download/v1.1.0/mpc.v1.1.0.x86_64-linux-gnu.tar.gz to /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/downloads/mpc.v1.1.0.x86_64-linux-gnu.tar.gz...
[13:31:57] #####################################################################[13:31:58] #####################################################################[13:31:58] ######                                                               [13:31:58] #######                                                              [13:31:58] #############                                                        [13:31:58] ##############                                                       [13:31:58] ####################                                                 [13:31:58] #####################                                                [13:31:58] ############################                                         [13:31:58] ############################                                         [13:31:58] ###################################                                  [13:31:58] ###################################                                  [13:31:58] ##########################################                           [13:31:58] ##########################################                           [13:31:58] #################################################                    [13:31:58] ##################################################                   [13:31:58] ########################################################             [13:31:58] #########################################################            [13:31:58] ################################################################     [13:31:58] ################################################################     [13:31:58] #####################################################################[13:31:58] #####################################################################[13:31:58] ######################################################################## 100.0%
Info: Found a valid dl path libgmp.so while looking for libmpfr
Info: Found a valid dl path libgmp.so.10 while looking for libmpfr
Info: Found a valid dl path libgmp.so.10.3.2 while looking for libmpfr
Info: Found a valid dl path libgmpxx.so while looking for libmpfr
Info: Found a valid dl path libgmpxx.so.4 while looking for libmpfr
Info: Found a valid dl path libgmpxx.so.4.5.2 while looking for libmpfr
Info: Found a valid dl path libmpc.so while looking for libmpfr
Info: Found a valid dl path libmpc.so.3 while looking for libmpfr
Info: Found a valid dl path libmpc.so.3.1.0 while looking for libmpfr
Info: Found a valid dl path libmpfr.so while looking for libmpfr
Info: /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libmpfr.so matches our search criteria of libmpfr
Info: Found a valid dl path libgmp.so while looking for libgmp
Info: /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libgmp.so matches our search criteria of libgmp
Info: Found a valid dl path libgmp.so while looking for libsymengine
Info: Found a valid dl path libgmp.so.10 while looking for libsymengine
Info: Found a valid dl path libgmp.so.10.3.2 while looking for libsymengine
Info: Found a valid dl path libgmpxx.so while looking for libsymengine
Info: Found a valid dl path libgmpxx.so.4 while looking for libsymengine
Info: Found a valid dl path libgmpxx.so.4.5.2 while looking for libsymengine
Info: Found a valid dl path libmpc.so while looking for libsymengine
Info: Found a valid dl path libmpc.so.3 while looking for libsymengine
Info: Found a valid dl path libmpc.so.3.1.0 while looking for libsymengine
Info: Found a valid dl path libmpfr.so while looking for libsymengine
Info: Found a valid dl path libmpfr.so.6 while looking for libsymengine
Info: Found a valid dl path libmpfr.so.6.0.1 while looking for libsymengine
Info: Found a valid dl path libsymengine.so while looking for libsymengine
Info: /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libsymengine.so matches our search criteria of libsymengine
Info: /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libsymengine.so cannot be dlopen'ed
Info: Found a valid dl path libsymengine.so.0.3 while looking for libsymengine
Info: /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libsymengine.so.0.3 matches our search criteria of libsymengine
Info: /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libsymengine.so.0.3 cannot be dlopen'ed
Info: Found a valid dl path libsymengine.so.0.3.0 while looking for libsymengine
Info: /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libsymengine.so.0.3.0 matches our search criteria of libsymengine
Info: /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libsymengine.so.0.3.0 cannot be dlopen'ed
Info: Could not locate libsymengine inside /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib
Info: Found a valid dl path libgmp.so while looking for libmpc
Info: Found a valid dl path libgmp.so.10 while looking for libmpc
Info: Found a valid dl path libgmp.so.10.3.2 while looking for libmpc
Info: Found a valid dl path libgmpxx.so while looking for libmpc
Info: Found a valid dl path libgmpxx.so.4 while looking for libmpc
Info: Found a valid dl path libgmpxx.so.4.5.2 while looking for libmpc
Info: Found a valid dl path libmpc.so while looking for libmpc
Info: /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libmpc.so matches our search criteria of libmpc
==============================[ ERROR: SymEngine ]==============================

LoadError: BinaryProvider.LibraryProduct(nothing, String["libsymengine"], :libsymengine, "Prefix(/data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3)") is not satisfied, cannot generate deps.jl!
while loading /data/ben/.julia/v0.6/SymEngine/deps/build.jl, in expression starting on line 41

================================================================================

================================[ BUILD ERRORS ]================================

WARNING: SymEngine had build errors.

 - packages with build errors remain installed in /data/ben/.julia/v0.6
 - build the package(s) and all dependencies with `Pkg.build("SymEngine")`
 - build a single package by running its `deps/build.jl` script

================================================================================
INFO: Package database updated
```

Here's my versioninfo:
```
Julia Version 0.6.4
Commit 9d11f62 (2018-07-09 19:09 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: AMD Opteron(tm) Processor 6386 SE
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Piledriver MAX_THREADS=16)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, bdver1)
```

**Comment by isuruf** [comment]
what does `Libdl.dlopen("/data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libsymengine.so")` give you?

**Comment by qua4tre** [comment]
ERROR: could not load library "/data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libsymengine.so"
/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /data/ben/.julia/v0.6/SymEngine/deps/symengine-0.3/lib/libsymengine.so)
Stacktrace:
 [1] dlopen(::String, ::UInt32) at ./libdl.jl:97 (repeats 2 times)

**Comment by isuruf** [comment]
Did you build julia from source or used your package manager or used the binaries from julia website?

**Comment by qua4tre** [comment]
Built from source.

**Comment by daxinnan** [comment]
Anythin solved? Problem occured in julia v1.0.1 again. My versioninfo is 
`Julia Version 1.0.1
Commit 0d713926f8* (2018-09-29 19:05 UTC)
Platform Info:
  OS: Linux (x86_64-redhat-linux)
  CPU: Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, broadwell)
`

**Comment by isuruf** [comment]
Did you build julia yourself? I think this is due to https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/230
